[docs] Document Nutanix machineset sample

yanhua121 opened the pull request:
[docs] Document Nutanix machineset sample
* Documents how a user can create a Windows MachineSet in a Nutanix cluster
* Adds pre-requisites to run Windows worker machines in a Nutanix cluster

yanhua121 commented:
/do-not-merge/work-in-progress

aravindhp commented:
/cc @DCChadwick

DCChadwick commented on docs/machineset-nutanix.md:
```diff
@@ -0,0 +1,63 @@
+# Creating a Nutanix Windows MachineSet
+
+_\<windows_server_image_name\>_ should be replaced with the full name of an image of a [supported version](wmco-prerequisites.md#supported-windows-server-versions) of Windows, pre-uploaded to the Prism-Central/Prism-Element where the Machine VMs will be created with.
```
```suggestion
Replace _\<windows_server_image_name\>_ with the full name of an image of a [supported version](wmco-prerequisites.md#supported-windows-server-versions) of Microsoft Windows that is pre-uploaded to the Prism-Central/Prism-Element where the Machine VMs will be created.
```

DCChadwick commented on docs/machineset-nutanix.md:
```diff
@@ -0,0 +1,63 @@
+_\<windows_server_image_name\>_ should be replaced with the full name of an image of a [supported version](wmco-prerequisites.md#supported-windows-server-versions) of Windows, pre-uploaded to the Prism-Central/Prism-Element where the Machine VMs will be created with.
+For example, `nutanix-windows-server-2022`
+
+_\<infrastructure_id\>_ should be replaced with the output of:
```
Replace _\<infrastructure_id\>_ with the output:

DCChadwick replied:
```suggestion
Replace _\<infrastructure_id\>_ with the output:
```

DCChadwick commented on docs/machineset-nutanix.md:
```diff
@@ -0,0 +1,63 @@
+# Creating a Nutanix Windows MachineSet
+
+_\<windows_server_image_name\>_ should be replaced with the full name of an image of a [supported version](wmco-prerequisites.md#supported-windows-server-versions) of Windows, pre-uploaded to the Prism-Central/Prism-Element where the Machine VMs will be created with.
```
I'm assuming that "Windows" on line 3 refers to a Microsoft Windows OS and not the Windows Machine Config Operator. To avoid any potential confusion, consider replacing "of Windows" with "of the Windows operating system".

yanhua121 replied:
Changed according to the suggestion.

DCChadwick commented on docs/nutanix-prerequisites.md:
```diff
@@ -0,0 +1,30 @@
+# Nutanix Platform Pre-Requisites
+
+In order to successfully use the Windows Machine Config Operator (WMCO) on a Nutanix Platform, 
```
```suggestion
To use the Windows Machine Config Operator (WMCO) on a Nutanix Platform, 
```

DCChadwick commented on docs/nutanix-prerequisites.md:
```diff
@@ -0,0 +1,30 @@
+# Nutanix Platform Pre-Requisites
+
+In order to successfully use the Windows Machine Config Operator (WMCO) on a Nutanix Platform, 
+the following pre-requisites are required:
```
Omit line 4. (See line 3.)

yanhua121 replied:
Done

DCChadwick commented on docs/nutanix-prerequisites.md:
```diff
@@ -0,0 +1,30 @@
+In order to successfully use the Windows Machine Config Operator (WMCO) on a Nutanix Platform, 
+the following pre-requisites are required:
+
+* Upload a [VM image with a compatible Windows Server version](wmco-prerequisites.md#supported-windows-server-versions) to the Prism-Central/Prism-Element where the Machine VMs will be created with.
```
```suggestion
* Upload a [VM image with a compatible Windows Server version](wmco-prerequisites.md#supported-windows-server-versions) to the Prism-Central/Prism-Element where the Machine VMs will be created.
```

DCChadwick commented on docs/nutanix-prerequisites.md:
```diff
@@ -0,0 +1,30 @@
+the following pre-requisites are required:
+
+* Upload a [VM image with a compatible Windows Server version](wmco-prerequisites.md#supported-windows-server-versions) to the Prism-Central/Prism-Element where the Machine VMs will be created with.
+* Add a [DNS entry](#adding-a-dns-entry-in-nutanix-environment) for the internal API endpoint in the Nutanix environment
```
```suggestion
* Add a [DNS entry](#adding-a-dns-entry-in-nutanix-environment) for the internal API endpoint in the Nutanix environment.
```

yanhua121 replied:
Fixed

DCChadwick commented on docs/nutanix-prerequisites.md:
```diff
@@ -0,0 +1,30 @@
+
+## Adding a DNS entry in Nutanix environment
+
+WMCO downloads the ignition files from the internal API server endpoint. To enable this, add a new DNS entry for
```
Add a noun after "this" so that the reader and any translators will know what "this" refers to. For example:

"To enable this download" or "To enable these files" or "To enable this feature"

DCChadwick commented on docs/nutanix-prerequisites.md:
```diff
@@ -0,0 +1,30 @@
+Note: The DNS entry can be a CNAME or an additional A record.
+
+The above DNS entry ensures that Windows VM can download the ignition file from the internal API server 
+and the `kubelet` on the configured VM can communicate with the internal API server. In the case of Linux nodes,
```
```suggestion
and the `kubelet` on the configured VM can communicate with the internal API server. For Linux nodes,
```

DCChadwick commented on docs/nutanix-prerequisites.md:
```diff
@@ -0,0 +1,30 @@
+
+The above DNS entry ensures that Windows VM can download the ignition file from the internal API server 
+and the `kubelet` on the configured VM can communicate with the internal API server. In the case of Linux nodes,
+CoreDNS is running on every node which helps in resolving the internal API server URL. The external API endpoint
```
```suggestion
CoreDNS runs on every node, which helps resolve the internal API server URL. The external API endpoint
```

yanhua121 replied:
Changed according to the suggestion.

DCChadwick reviewed (commented):
I've added some suggested changes to the docs/machineset-nutanix.md and the docs/nutanix-prerequisites.md files. If prompted, avoid using the "Commit suggestion" button. Instead, if you like the suggestions, make changes manually.

yanhua121 commented on docs/nutanix-prerequisites.md:
```diff
@@ -0,0 +1,30 @@
+# Nutanix Platform Pre-Requisites
+
+In order to successfully use the Windows Machine Config Operator (WMCO) on a Nutanix Platform, 
```
Changed according to the suggestion.

yanhua121 commented:
> I've added some suggested changes to the docs/machineset-nutanix.md and the docs/nutanix-prerequisites.md files.

@DCChadwick Thanks for the review comments. I addressed them all.

aravindhp reviewed (commented):
Thanks for doing this, @yanhua121

yanhua121 commented on docs/wmco-prerequisites.md:
```diff
@@ -31,6 +32,7 @@ these errors, only use the appropriate version according to the cloud provider i
 | Azure          | - Windows Server 2019, version 1809 Long-Term Servicing Channel (LTSC)<br>- Windows Server 2022 Long-Term Servicing Channel (LTSC)|
 | GCP            | Windows Server 2022 Long-Term Servicing Channel (LTSC)                                                                            |
 | VMware vSphere | Windows Server 2022 Long-Term Servicing Channel (LTSC)                                                                            |
+| Nutanix        | Windows Server 2022 Long-Term Servicing Channel (LTSC)                                                                            |
```
Similar to VMware, we cannot support Windows server 2019. According to Wolfgang, mainly due to the customVXLan Port issue with the Windows 2019.

aravindhp commented:
We can get this merged, once @DCChadwick gives a lgtm

yanhua121 commented:
@DCChadwick Coul gives a lgtm

> > I've added some suggested changes to the docs/machineset-nutanix.md and the docs/nutanix-prerequisites.md files.
> 
> @DCChadwick Thanks for the review comments. I addressed them all.

@DCChadwick Could you please take another look at this PR and see if you can give a "lgtm" label for it to merge? Thanks.

sebsoto commented:
LGTM but I'll wait until @DCChadwick gives the `/lgtm`

DCChadwick commented on docs/machineset-nutanix.md:
```diff
@@ -0,0 +1,63 @@
+Replace _\<windows_server_image_name\>_ with the full name of an image of a [supported version](wmco-prerequisites.md#supported-windows-server-versions) of the Windows operating system that is pre-uploaded to the Prism-Central/Prism-Element where the Machine VMs will be created.
+For example, `nutanix-windows-server-2022`
+
+Replace _\<infrastructure_id\>_ should be replaced with the output:
```
```suggestion
Replace _\<infrastructure_id\>_ with the output:
```

yanhua121 replied:
Fixed.

DCChadwick commented:
I just had one suggestion. (See comment above.) Otherwise, this looks good to me.

yanhua121 commented:
> I just had one suggestion. (See comment above.) Otherwise, this looks good to me.

@DCChadwick Fixed. Can you take another look? Thanks!

DCChadwick commented:
/label docs-approved

aravindhp reviewed (approved):
/lgtm
